### PR TITLE
Throttle refactor

### PR DESCRIPTION
**Pull request by aim2120**
## Description

Refactors the throttle operation to behave more similarly to Combine's.

Also improves the logic / performance of the operator by using fewer Tasks.

Note: while swift-async-algorithms does have their own throttle operator, there has been much discussion around undesirable behavior of the operator. See [this Swift forums thread](https://forums.swift.org/t/surprising-semantics-of-throttled-sequences/65409) and [this GitHub issue](https://github.com/apple/swift-async-algorithms/issues/266). I've mostly ignored the design of swift-async-algorithms' implementation in favor of attempting to replicate Combine's.

I've also improved the throttle DSL tests. There was a decent amount of logic attempting to "figure out" what the elements should be at a given step, which complicated the test harness and was prone to introducing errors itself. I've chosen to adjust the DSL to be more similar to the [validation used by swift-async-algorithms](https://github.com/apple/swift-async-algorithms/blob/4c3ea81f81f0a25d0470188459c6d4bf20cf2f97/Sources/AsyncSequenceValidation/AsyncSequenceValidation.docc/Validation.md), which uses alignment to determine what elements should be present at each validation point.

Checks:
- [x] Successfully ran `ThrottleSequenceTests` x1000

### CODE REVIEW

**Comment by aim2120** [comment]
CI failure was unrelated to change:
```
✘ Test delay_DelaysCorrectlyEvenAfterIntervalHasPassed() recorded an issue at DelaySequenceTests.swift:118:25: Expectation failed: (elapsedTime → 0.02 seconds) >= (delayDuration + .milliseconds(15) → 0.025 seconds)
```

**Comment by Tyler-Keith-Thompson** [comment]
First off, thank you so much for the PR. This is sorely needed. It's going to take me a while to review since it's a nice and complicated operator. I'll also want to sanity check it against some Combine output in a playground.

**Review by Tyler-Keith-Thompson — APPROVED** [review]
I've put it through its paces and compared 1:1 with Combine. So far the same elements are emitted in the same order. These timing type operators are always tricky. However, the test suite is robust. I think we're good to proceed!

File: Tests/AfluentTests/SequenceTests/ThrottleSequenceTests.swift [hunk 1]
```diff
@@ -65,156 +58,429 @@ struct ThrottleSequenceTests {
...
+            Output: "\(expectedOutput)"
+            """)
+
+        let advancedDuration = ManagedAtomic<Int>(0)
+
         for (i, step) in streamInput.enumerated() {
             if step == "-" {
                 await testClock.advance(by: .milliseconds(10))
-                await Task.yield()
+                await Task.megaYield()
```
[the remark this answers is not in the document]
Comment (Tyler-Keith-Thompson) [reply Tests/AfluentTests/SequenceTests/ThrottleSequenceTests.swift:438]: The tests as written (thanks to your megaYield) seem to run reliably on my machine as well. That is good enough for me!

**Pull request merged**